src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -498,12 +504,19 @@ private Task GetJoiningPreconditionPromise(MembershipTableData table)
             // only process the table if in the active or ShuttingDown state. In other states I am not ready yet.
             if (IsFunctionalMBR(CurrentStatus))
             {
-                foreach (var entry in table.Members.Select(tuple => tuple.Item1).Where(
-                    item => !item.SiloAddress.Endpoint.Equals(MyAddress.Endpoint)))
+                foreach (var entry in table.Members.Select(tuple => tuple.Item1))
                 {
-                    bool changed = membershipOracleData.TryUpdateStatusAndNotify(entry);
-                    localViewChanged = localViewChanged || changed;
+                    if (!entry.SiloAddress.Endpoint.Equals(MyAddress.Endpoint))
+                    {
+                        bool changed = membershipOracleData.TryUpdateStatusAndNotify(entry);
+                        localViewChanged = localViewChanged || changed;
+                    }
+                    else
+                    {
+                        membershipOracleData.UpdateMyFaultAndUpdateZone(entry);
```

Comment:
I don't understand how this data can change while we are already running. Shouldn't we capture it once - when silo starts?